analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -91,30 +90,31 @@ private static void CheckConstructorDeclaration(SonarSyntaxNodeReportingContext
 
         private static bool IsConstructorRedundant(ConstructorDeclarationSyntax constructorDeclaration, SemanticModel semanticModel) =>
             constructorDeclaration is { ParameterList.Parameters.Count: 0, Body.Statements.Count: 0 }
+            && !IsStructWithInitializedFieldOrProperty((BaseTypeDeclarationSyntax)constructorDeclaration.Parent, semanticModel)
             && (IsSinglePublicConstructor(constructorDeclaration, semanticModel)
                 || constructorDeclaration.Modifiers.Any(SyntaxKind.StaticKeyword));
 
+        private static bool IsStructWithInitializedFieldOrProperty(BaseTypeDeclarationSyntax parentType, SemanticModel semanticModel) =>
+            parentType.Kind() is SyntaxKind.StructDeclaration or SyntaxKindEx.RecordStructDeclaration
+            && semanticModel.GetDeclaredSymbol(parentType) is { } typeSymbol
+            && typeSymbol.GetMembers().Any(f => f.Kind is SymbolKind.Field or SymbolKind.Property
+                                                      && f.DeclaringSyntaxReferences.Any(d => d.GetSyntax().GetInitializer() is not null));
+
         private static bool IsSinglePublicConstructor(ConstructorDeclarationSyntax constructorDeclaration, SemanticModel semanticModel) =>
             constructorDeclaration.Modifiers.Any(SyntaxKind.PublicKeyword)
             && IsInitializerEmptyOrRedundant(constructorDeclaration.Initializer)
-            && TypeHasExactlyOneConstructor(constructorDeclaration, semanticModel);
+            && TypeHasExactlyOneConstructor((BaseTypeDeclarationSyntax)constructorDeclaration.Parent, semanticModel);
```

Comment:
Consider making it a bit more robust
```suggestion
            && constructorDeclaration is { Parent: BaseTypeDeclarationSyntax typeDeclaration }
            && TypeHasExactlyOneConstructor(typeDeclaration, semanticModel);
```